onnxruntime/test/framework/kernel_registry_test.cc

```diff
@@ -90,5 +90,15 @@ TEST(KernelRegistryTests, two_versions3) {
   function_table.emplace_back(KernelDefBuilder().MayInplace(0, 0).TypeConstraint("T", DataTypeImpl::GetTensorType<float>()).SetName("Elu").SetDomain("").SinceVersion(6).Provider(kCpuExecutionProvider).Build());
   function_table.emplace_back(KernelDefBuilder().MayInplace(0, 0).TypeConstraint("T", DataTypeImpl::GetTensorType<float>()).SetName("Elu").SetDomain("").SinceVersion(1).Provider(kCpuExecutionProvider).Build());
   Status st;
+  ASSERT_TRUE((st = RegKernels(r, function_table, CreateFakeKernel)).IsOK()) << st.ErrorMessage();
```

Comment:
Just wondering, what if the model is stamped with opset 2,3,4,5,7?  Would it always find the correct kernel?

Reply:
It should find the right kernel, or fail if the right kernel doesn't exist.  

Say that schema for this operator exist in opsets 1, 4, and 6.  This will result in the second kernel registration (1, int_max) being used for models that import opset 1 through 3.  Models importing opsets 4 and 5 would fail given only the above two registrations.

A separate bug can currently cause the wrong kernel (and schema) to be used in models importing opsets beyond the maximum registered schema.  Winml currently uses a fix for this which is isolated to its branches.  See:
https://github.com/microsoft/onnxruntime/pull/2364/files

Reply:
My understanding is: Models importing opsets 4 and 5 shoudn't fail.

Reply:
Let's use the Elu as an example.  It has two versions: the initial one in opset 1 and later on it get updated in opset 6. 

So, a model is stamped with any opset should be able to find the op.
opset 1,2,3,4,5 -> Elu-1
others: -> Elu-6

Reply:
That is also my understanding.  In my example the schema was updated in opset 4, although it is different from the real ELU.  This explains the difference.